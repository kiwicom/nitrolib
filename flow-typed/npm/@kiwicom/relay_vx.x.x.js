// flow-typed signature: b1dc49e6afa4f11fce2e6c29369e7ae5
// flow-typed version: <<STUB>>/@kiwicom/relay_v3.1.1/flow_v0.102.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@kiwicom/relay'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@kiwicom/relay' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@kiwicom/relay/bin/kiwicom-relay-compiler' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/commitLocalUpdate' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/commitMutation' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/compiler/__generated__/persistOperationMutation.graphql' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/compiler/buildLanguagePlugin' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/compiler/formatGeneratedModule' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/compiler/index' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/compiler/persistOperation' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/createEnvironment' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/createFragmentContainer' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/createPaginationContainer' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/createRefetchContainer' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/createRequestHandler' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/fetchers/createNetworkFetcher' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/graphql' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/helpers' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/index' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/QueryRenderer' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/requestSubscription' {
  declare module.exports: any;
}

declare module '@kiwicom/relay/src/types.flow' {
  declare module.exports: any;
}

// Filename aliases
declare module '@kiwicom/relay/bin/kiwicom-relay-compiler.js' {
  declare module.exports: $Exports<'@kiwicom/relay/bin/kiwicom-relay-compiler'>;
}
declare module '@kiwicom/relay/src/commitLocalUpdate.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/commitLocalUpdate'>;
}
declare module '@kiwicom/relay/src/commitMutation.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/commitMutation'>;
}
declare module '@kiwicom/relay/src/compiler/__generated__/persistOperationMutation.graphql.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/compiler/__generated__/persistOperationMutation.graphql'>;
}
declare module '@kiwicom/relay/src/compiler/buildLanguagePlugin.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/compiler/buildLanguagePlugin'>;
}
declare module '@kiwicom/relay/src/compiler/formatGeneratedModule.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/compiler/formatGeneratedModule'>;
}
declare module '@kiwicom/relay/src/compiler/index.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/compiler/index'>;
}
declare module '@kiwicom/relay/src/compiler/persistOperation.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/compiler/persistOperation'>;
}
declare module '@kiwicom/relay/src/createEnvironment.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/createEnvironment'>;
}
declare module '@kiwicom/relay/src/createFragmentContainer.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/createFragmentContainer'>;
}
declare module '@kiwicom/relay/src/createPaginationContainer.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/createPaginationContainer'>;
}
declare module '@kiwicom/relay/src/createRefetchContainer.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/createRefetchContainer'>;
}
declare module '@kiwicom/relay/src/createRequestHandler.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/createRequestHandler'>;
}
declare module '@kiwicom/relay/src/fetchers/createNetworkFetcher.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/fetchers/createNetworkFetcher'>;
}
declare module '@kiwicom/relay/src/graphql.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/graphql'>;
}
declare module '@kiwicom/relay/src/helpers.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/helpers'>;
}
declare module '@kiwicom/relay/src/index.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/index'>;
}
declare module '@kiwicom/relay/src/QueryRenderer.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/QueryRenderer'>;
}
declare module '@kiwicom/relay/src/requestSubscription.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/requestSubscription'>;
}
declare module '@kiwicom/relay/src/types.flow.js' {
  declare module.exports: $Exports<'@kiwicom/relay/src/types.flow'>;
}
