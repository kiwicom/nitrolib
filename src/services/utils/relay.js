// @flow strict
import { Environment, Network, RecordSource, Store } from "relay-runtime";

export type Input = {
  query: string,
  variables: { [key: string]: string },
};

export type Call = (input: Input) => Promise<$FlowFixMe>;

const makeFetchQuery = (call: Call) => (operation, variables) =>
  call({
    query: operation.text,
    variables,
  });

const store = new Store(new RecordSource());

export const makeCall = (headers: { [string]: string }) => (input: Input) =>
  fetch("https://graphql.kiwi.com", {
    method: "POST",
    headers: {
      ...headers,
      "Content-type": "application/json",
      Accept: "application/json",
      "X-Client": "nitro",
    },
    body: JSON.stringify(input),
  }).then(res => res.json());

export const makeEnvironment = (call: Call) =>
  new Environment({
    network: Network.create(makeFetchQuery(call)),
    store,
  });
