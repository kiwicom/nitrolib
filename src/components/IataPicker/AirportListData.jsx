// @flow strict
import * as React from "react";
import { graphql, QueryRenderer } from "@kiwicom/relay";
import Alert from "@kiwicom/orbit-components/lib/Alert";

import environmentReal from "../../services/environment";
import AirportList from "./AirportList";
import Translate from "../Translate";

type Props = {|
  value: string,
  env: typeof environmentReal,
  onSelect: (id: string) => void,
|};

const AirportListData = ({ value, onSelect, env }: Props) => (
  <QueryRenderer
    environment={env}
    query={graphql`
      query AirportListDataQuery($input: String!) {
        allLocations(search: $input, options: { locationType: airport }) {
          ...AirportList_list
        }
      }
    `}
    variables={{ input: value }}
    onSystemError={() => (
      <Alert type="critical">
        <Translate t="common.api_error" />
      </Alert>
    )}
    onLoading={() => null}
    onResponse={res => <AirportList list={res.allLocations} onSelect={onSelect} />}
  />
);

AirportListData.defaultProps = {
  env: environmentReal,
};

export default AirportListData;
