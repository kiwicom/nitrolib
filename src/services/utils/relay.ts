import { createEnvironment, createNetworkFetcher } from "@kiwicom/relay";

type Headers = { [key: string]: string };

const makeEnvironment = (headers?: Headers, query: string = "https://graphql.kiwi.com") =>
  createEnvironment({
    fetchFn: createNetworkFetcher(query, {
      ...headers,
      "Content-type": "application/json",
      Accept: "application/json",
      "X-Client": "nitro",
    }),
  });

export default makeEnvironment;
