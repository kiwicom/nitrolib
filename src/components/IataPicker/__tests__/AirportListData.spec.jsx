// @flow strict
import * as React from "react";
import { mount } from "enzyme";
import { RelayEnvironmentProvider } from "@kiwicom/relay";
import type { Environment } from "@kiwicom/relay";
import { createMockEnvironment, MockPayloadGenerator } from "relay-test-utils";

import AirportListData from "../AirportListData";

const renderAirportListData = (environment: Environment) => (
  <RelayEnvironmentProvider environment={environment}>
    <AirportListData value="VIE" onSelect={jest.fn()} />
  </RelayEnvironmentProvider>
);

describe("#AirportListData", () => {
  test("render error", async () => {
    const environment = createMockEnvironment();
    const wrapper = mount(renderAirportListData(environment));

    environment.mock.rejectMostRecentOperation(new Error("error"));

    wrapper.update();

    expect(wrapper.find("ReactRelayQueryRenderer").exists()).toBe(true);
  });

  test("render loading", () => {
    const wrapper = mount(renderAirportListData(createMockEnvironment()));

    expect(wrapper.text()).toBe("");
  });

  test("render results", async () => {
    const environment = createMockEnvironment();
    const wrapper = mount(renderAirportListData(environment));

    environment.mock.resolveMostRecentOperation(operation =>
      MockPayloadGenerator.generate(operation, {
        Location() {
          return {
            locationId: "VIE",
            city: {
              name: "Vienna",
            },
          };
        },
      }),
    );

    wrapper.update();

    expect(
      wrapper
        .find("AirportResult__Name")
        .children()
        .text(),
    ).toBe("Vienna (VIE)");
  });
});
