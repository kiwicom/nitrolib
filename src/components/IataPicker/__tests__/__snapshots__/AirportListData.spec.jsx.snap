// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#AirportListData render error 1`] = `"<div class=\\"Alert__StyledAlert-svgppc-0 jCMjFD\\"><div class=\\"Alert__ContentWrapper-svgppc-2 gGclWN\\"><div class=\\"Alert__Content-svgppc-4 cGbCFo\\">common.api_error</div></div></div>"`;

exports[`#AirportListData render loading 1`] = `null`;

exports[`#AirportListData render results 1`] = `"<button class=\\"AirportResult__Container-sc-1cbq8wi-0 iktDbi\\"><div class=\\"CountryFlag__StyledCountryFlag-h98fwe-0 RxxyG\\"><img src=\\"//images.kiwi.com/flags/24x0/flag-at.jpg\\" srcset=\\"//images.kiwi.com/flags/48x0/flag-at.jpg 2x\\" class=\\"CountryFlag__StyledImage-h98fwe-1 iYSrFt\\"><div class=\\"CountryFlag__StyledShadow-h98fwe-2 cKAvsc\\"></div></div><span class=\\"AirportResult__Name-sc-1cbq8wi-1 gMohdB\\">Vienna (VIE)</span></button>"`;
